packages/data-point/lib/entity-types/base-entity/resolve.test.js

```diff
@@ -220,6 +220,27 @@ describe('ResolveEntity.resolveEntity', () => {
       })
   })
 
+  test('outputType - throws error if outputType does not pass when after changes output', () => {
+    return resolveEntity('model:c.4', 'some string')
+      .catch(e => e)
+      .then(e => {
+        expect(e).toMatchSnapshot()
+      })
+  })
+
+  test('outputType - throws error if outputType does not pass when model:after changes output', () => {
+    dataPoint.middleware.use('model:after', (acc, next) => {
+      acc.resolve(1)
+      next(null)
+    })
+
+    return resolveEntity('model:c.1', 'some string')
+      .catch(e => e)
+      .then(e => {
+        expect(e).toMatchSnapshot()
+      })
+  })
```

Comment:
we should also have a similar test for the "after" middleware that's executed for all entities:

```js
dataPoint.middleware.use('model:after', /* ... */
```

Reply:
This is for the `model:after`. Did you mean `model:before`?

Reply:
sorry, copied and pasted and forgot to remove the "model" key:
```js
dataPoint.middleware.use('after', /* ... */
```

Reply:
Cool. Yes, that'd be good to test for too. Also going to add comparable tests for the `before` versions of these lifecycle hooks.